cli/azd/internal/agent/conversational_agent.go

```diff
@@ -124,8 +127,9 @@ func (aai *ConversationalAzdAiAgent) SendMessage(ctx context.Context, args ...st
 	return output, nil
 }
 
-func (aai *ConversationalAzdAiAgent) renderThoughts(ctx context.Context) (func(), error) {
+func (aai *ConversationalAzdAiAgent) renderThoughts(ctx context.Context) (func(), chan struct{}, error) {
 	var latestThought string
+	completionChan := make(chan struct{})
```

Comment:
I don't think we should need another channel and have to manage closing this outside of the function.  In Go the channel should be closed by the creator. In this case the channel is created in the function and expected to be close outside.

In our case we have the incoming context that is "cancelled" when we are done.  We likely can use this to handle the last action printing vs needing another special channel.